driver/src/contracts/stablex_contract.rs

```diff
@@ -108,20 +114,45 @@ impl StableXContract for StableXContractImpl {
         Ok(Duration::from_secs(remaining_seconds.as_u64()))
     }
 
+    fn get_last_block_for_batch(&self, batch_id: u32) -> Result<Option<u64>> {
+        const BATCH_DURATION: u64 = 300;
+
+        let web3 = self.instance.raw_instance().web3();
+        let get_block = |block_number: BlockNumber| -> Result<_> {
+            web3.eth()
+                .block(block_number.into())
+                .wait()?
+                .ok_or_else(|| anyhow!("block {:?} is missing", block_number))
+        };
+
+        let mut current_block = get_block(BlockNumber::Latest)?;
+        let mut block_number = None;
+        while (batch_id as u64) < current_block.timestamp.as_u64() / BATCH_DURATION
+            && current_block.number != Some(0.into())
+        {
+            let previous_block_number = current_block.number.ok_or_else(|| {
+                anyhow!("block {:?} has a missing block number", current_block.hash)
+            })? - 1;
```

Comment:
I feel like this is a classic interview question, and my hunch would be that the optimal strategy involves doubling the gap and then doing a binary search when we overshot the target block time.
Not sure this is necessary here though since we should always be pretty close to the needle.

Reply:
I think that's a great idea, I would implement it as a separate PR though.

Reply:
I’m not sure what exactly the interview problem coming from this is meant to be? I can only assume it has something to do with getting blocks.

Reply:
Its kind of doing a binary search on the block :smile:

Reply:
@fleupold Where the awesome search strategy?! :stuck_out_tongue_closed_eyes:

Reply:
I didn't want to take the fun stuff away from you 😉 More seriously, this is probably something we want to test and thus requires some more abstraction which would go beyond the scope of this PR.

Reply:
I see, the binary search is the "way we should do it" but for now we will simply step backwards one at a time.